airflow/providers/databricks/hooks/databricks.py

```diff
@@ -522,6 +515,20 @@ def uninstall(self, json: dict) -> None:
         """
         self._do_api_call(UNINSTALL_LIBS_ENDPOINT, json)
 
+    @staticmethod
+    def _is_aad_token_valid(aad_token: dict) -> bool:
```

Comment:
why do we need a separate function that is called from one place?

Reply:
- Mainly for readability to hide the details for checking that token is valid under the function, because it's not the main purpose of the parent function `_get_aad_token`
- There's a mistake in the current function implementation: it subtracts `TOKEN_REFRESH_LEAD_TIME` out of the current time, while it should actually sum it. With this we fix it and cover the function with tests.